core/src/test/scala/integration/kafka/api/PlaintextProducerSendTest.scala

```diff
@@ -41,6 +42,42 @@ class PlaintextProducerSendTest extends BaseProducerSendTest {
     producer.send(record)
   }
 
+  @Test
+  def testRecordHeaders(): Unit = {
+    val producerProps = new Properties()
+    producerProps.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, brokerList)
+    producerProps.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.StringSerializer")
+    producerProps.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.StringSerializer")
+    val producer = new KafkaProducer[String, String](producerProps)
+    var recordHeaders = new RecordHeaders()
+    recordHeaders.add("RecordHeaderKey", "RecordHeaderValue".getBytes)
```

Comment:
Can you add a comment why this is invalid? Also maybe rename it to invalidRecordHeaders or unsupportedRecordHeaders.

Also need a todo to delete this test when v2 format is supported.

Reply:
I have already added it at the beginning of MemoryRecordsBuilder.java -- We can revert the whole commit by using that as reference.